pkgimage.mk

```diff
@@ -24,129 +24,5 @@ $(JULIA_DEPOT_PATH):
-$(eval $(call stdlib_builder,LibUnwind_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,MbedTLS_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,nghttp2_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,OpenLibm_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,PCRE2_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,Zlib_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,dSFMT_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,libLLVM_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,libblastrampoline_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,p7zip_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,OpenBLAS_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,Markdown,Base64))
-$(eval $(call stdlib_builder,Printf,Unicode))
-$(eval $(call stdlib_builder,Random,SHA))
-$(eval $(call stdlib_builder,Logging,StyledStrings))
-$(eval $(call stdlib_builder,Tar,ArgTools,SHA))
-$(eval $(call stdlib_builder,DelimitedFiles,Mmap))
-$(eval $(call stdlib_builder,JuliaSyntaxHighlighting,StyledStrings))
-
-# 2-depth packages
-$(eval $(call stdlib_builder,LLD_jll,Zlib_jll libLLVM_jll Artifacts Libdl))
-$(eval $(call stdlib_builder,LibSSH2_jll,Artifacts Libdl MbedTLS_jll))
-$(eval $(call stdlib_builder,MPFR_jll,Artifacts Libdl GMP_jll))
-$(eval $(call stdlib_builder,LinearAlgebra,Libdl libblastrampoline_jll OpenBLAS_jll))
-$(eval $(call stdlib_builder,Dates,Printf))
-$(eval $(call stdlib_builder,Distributed,Random Serialization Sockets))
-$(eval $(call stdlib_builder,Future,Random))
-$(eval $(call stdlib_builder,UUIDs,Random SHA))
-$(eval $(call stdlib_builder,InteractiveUtils,Markdown))
-
- # 3-depth packages
-$(eval $(call stdlib_builder,LibGit2_jll,MbedTLS_jll LibSSH2_jll Artifacts Libdl))
-$(eval $(call stdlib_builder,LibCURL_jll,LibSSH2_jll nghttp2_jll MbedTLS_jll Zlib_jll Artifacts Libdl))
-$(eval $(call stdlib_builder,REPL,InteractiveUtils Markdown Sockets StyledStrings Unicode))
-$(eval $(call stdlib_builder,SharedArrays,Distributed Mmap Random Serialization))
-$(eval $(call stdlib_builder,TOML,Dates))
-$(eval $(call stdlib_builder,Test,Logging Random Serialization InteractiveUtils))
-
-# 4-depth packages
-$(eval $(call stdlib_builder,LibGit2,LibGit2_jll NetworkOptions Printf SHA Base64))
-$(eval $(call stdlib_builder,LibCURL,LibCURL_jll MozillaCACerts_jll))
-
-# 5-depth packages
-$(eval $(call stdlib_builder,Downloads,ArgTools FileWatching LibCURL NetworkOptions))
-
-# 6-depth packages
-$(eval $(call stdlib_builder,Pkg, Artifacts Dates Downloads FileWatching LibGit2 Libdl\
-								  Logging Markdown Printf REPL Random SHA Serialization\
-								  TOML Tar UUIDs p7zip_jll))
-
-# 7-depth packages
-$(eval $(call stdlib_builder,LazyArtifacts,Artifacts Pkg))
-$(eval $(call stdlib_builder,REPLExt,Pkg REPL))
-
-$(eval $(call stdlib_builder,SparseArrays,Libdl LinearAlgebra Random Serialization SuiteSparse_jll))
-$(eval $(call stdlib_builder,Statistics,LinearAlgebra SparseArrays))
+all-release all-debug:
```

Comment:
Things we are losing here right now.

`stdlib_builder` had a dependency on the source files, the Project.toml and the version file for the stdlib.

Similar to `SYSIMG_STDLIB_SRCS` we will need `STDLIB_SRCS` for the independent stdlibs and use that as a condition to this build step.

Also it's probably better to split this in to two and have a marker file instead of a phony target, so that make can skip running this step when all the preconditions are up to date.

Reply:
Another precondition needs to be `$(build_private_libdir)/sys.$(SHLIB_EXT)`

Reply:
Calling `precompilepkgs` is a noop if there's nothing to do. That's one of its strengths. I think we just want to call it every time? The only thing thats awkward is that currently make will print this for a noop
```
julia % make -j6
    JULIA all-release
julia % 
```